custom_components/meural/config_flow.py

```diff
@@ -41,12 +41,15 @@ async def async_step_user(self, user_input=None):
         errors = {}
         if user_input is not None:
             try:
-                token = await validate_input(self.hass, user_input)
+                await validate_input(self.hass, user_input)
 
                 await self.async_set_unique_id(user_input["email"], raise_on_progress=False)
                 return self.async_create_entry(
                     title=user_input["email"],
-                    data={"email": user_input["email"], "token": token},
+                    data={
```

Comment:
What I dislike about this new approach is that the token is not stored at all, even if it is valid for 3 months! It would be nice if we can still store the token in the config entry data and restore it at startup.

You would then need to pass a callback function to PyMeural to be called when the token is updated. 

```python
# inside async_setup_entry
entry = ...

def token_updated(token):
    hass.config_entries.async_update_entry(entry, data={**entry.data, "token": token})
```